### PR TITLE
Getting started k8s part3

### PR DESCRIPTION
**Pull request by spara**

### CODE REVIEW

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+![Architecture]()
+
+Let’s look at the Orders microservice, which is made up of a service worker and MongoDB for a backing service. Note that the code is organized by microservice with a separate Deployment and Service for each service and backing resource. The ordersDb configuration is straightforward, of note are the options in the container spec where we set [securityContext](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) and a [emptyDir volume](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) for storage. An emptyDir volume persists even if the container crashes, and the data in the volume is available if the container is restarted. The service definition for ordersDbService is also simple and opens port:27017 and ties the service back to the Deployment by the *orders-db* label.
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:35]:
```suggestion
Let’s look at the Orders microservice, which is made up of a service worker and MongoDB for a backing service. Note that the code is organized by microservice with a separate Deployment and Service for each service and backing resource. The ordersDb configuration is straightforward, of note are the options in the container spec where we set [securityContext](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) and a [emptyDir volume](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) for storage. An emptyDir volume persists even if the container crashes, and the data in the volume is available if the container is restarted. The service definition for ordersDbService is also simple and opens `port:27017` and ties the service back to the Deployment by the *orders-db* label.
```

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+                port: 27017,
+                targetPort: 27017,
+            },
+        ],
+        selector: ordersDb.spec.template.metadata.labels,
+    },
+});
+```
+
+The orders Deployment is similar to the ordersDb Deployment, but it includes environmental variables, such as JAVA_OPTS to set heap memory size and garbage collection intervals, for the containerized application. The container runs as a non-root user and drops all capabilities that can run as a non-root user, except for NET_BIND_SERVICE, which lets the container bind to any port.
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:122]:
```suggestion
The orders Deployment is similar to the ordersDb Deployment, but it includes environmental variables, such as `JAVA_OPTS` to set heap memory size and garbage collection intervals, for the containerized application. The container runs as a non-root user and drops all capabilities that can run as a non-root user, except for `NET_BIND_SERVICE`, which lets the container bind to any port.
```

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+    allocateIpAddress: true,
+    isMinikube: config.getBoolean("isMinikube"),
+});
+
+export let frontendIp = frontend.ipAddress;
+```
+
+The ServiceDeployment class is defined in the *k8sjs* module using ComponentResource. We define deployment, service, and ipAddress as public read-only properties. The constructor takes three arguments:
+
+name: The unique name of the resource.
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:256]:
```suggestion
`name`: The unique name of the resource.
```

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+args: Information passed to [initialize] method.
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:257]:
```suggestion
`args`: Information passed to [initialize] method.
```

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+opts: A bag of options that control this resource's behavior.
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:258]:
```suggestion
`opts`: A bag of options that control this resource's behavior.
```

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+// Export the public IP for Wordpress.
+const frontend = wordpress.getResourceProperty("v1/Service", "wpdev-wordpress", "status");
+export const frontendIp = frontend.apply(status => status.loadBalancer.ingress[0].ip);
+```
+
+## Choosing how to deploy
+
+Using a Helm chart to deploy an application is the simplest solution if it’s a standard application. If you are deploying your application, you might start with a boilerplate code defining Deployments, Services, and other objects in a prototype and later refactor it using Components. These methods aren’t exclusive; for example, you could use a Helm chart to deploy an [ELK stack](https://www.elastic.co/what-is/elk-stack) to gather metrics from your application and display them using [Kibana](https://www.elastic.co/kibana). As an application becomes more complex, e.g., every function is a microservice, it becomes more unwieldy and difficult to manage. Infrastructure as code can help reduce that complexity through abstracting resources into higher-level objects, applying software development practices to infrastructure, and enabling robust and reliable deployments.
+
+## Conclusion
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:365]: Consider adding a CTA

Comment (spara) [reply content/blog/getting-started-with-k8s-part3/index.md:365]: will add

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 7]
```diff
@@ -365,3 +366,9 @@ Using a Helm chart to deploy an application is the simplest solution if it’s a
 ## Conclusion
 
 This article demonstrated three ways to deploy an application in Kubernetes. In the next installment, we’ll examine how to maintain your application with [configmaps](https://kubernetes.io/docs/concepts/configuration/configmap/), perform rollouts and different deployment strategies, adopting existing resources, and more.
+
+This is the second in a series of articles on using infrastructure as code for Kubernetes. In the next article, we'll examine a complex application deployment and a deployment using a Helm chart. Until the next post, learn more about Kubernetes with these resources.
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md]: This refs the 2nd part of the series, but the PR is for part 3

Comment (spara) [reply content/blog/getting-started-with-k8s-part3/index.md]: Odd, local repo has the changes. Pushing again.

**Pull request closed without merging**